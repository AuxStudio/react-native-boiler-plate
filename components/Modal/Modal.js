import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import Modal from 'react-native-modal';

const propTypes = {
  isVisible: PropTypes.bool,
  handleClose: PropTypes.func,
  children: PropTypes.node.isRequired,
};

const defaultProps = {};

const ModalComponent = ({ isVisible, handleClose, children }) => {
  return (
    <View>
      <Modal
        isVisible={isVisible}
        onBackdropPress={handleClose}
        onSwipe={handleClose}
        swipeDirection="down"
        hideModalContentWhileAnimating
      >
        {children}
      </Modal>
    </View>
  );
};

ModalComponent.propTypes = propTypes;
ModalComponent.defaultProps = defaultProps;

export default ModalComponent;
