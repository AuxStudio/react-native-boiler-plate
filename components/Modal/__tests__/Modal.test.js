import React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';

import Modal from '..';

describe('Modal', () => {
  it('renders with all props', () => {
    expect(
      renderer.create(
        <Modal isVisible handleClose={jest.fn()}>
          <View />
        </Modal>,
      ),
    ).toMatchSnapshot();
  });

  it('renders with minimum required props', () => {
    expect(
      renderer.create(
        <Modal>
          <View />
        </Modal>,
      ),
    ).toMatchSnapshot();
  });
});
