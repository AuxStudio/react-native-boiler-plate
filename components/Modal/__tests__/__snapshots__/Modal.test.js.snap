// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders with all props 1`] = `
<View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={true}
    onModalHide={[Function]}
    onRequestClose={[Function]}
    onSwipe={[MockFunction]}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    swipeDirection="down"
    swipeThreshold={100}
    transparent={true}
    visible={true}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      hideModalContentWhileAnimating={true}
      onModalHide={[Function]}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      onSwipe={[MockFunction]}
      pointerEvents="box-none"
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "left": 0,
          "margin": 37.5,
          "top": 0,
          "transform": Array [
            Object {
              "translateY": 1334,
            },
          ],
        }
      }
      swipeDirection="down"
      swipeThreshold={100}
    >
      <View />
    </View>
  </Modal>
</View>
`;

exports[`Modal renders with minimum required props 1`] = `
<View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={true}
    onModalHide={[Function]}
    onRequestClose={[Function]}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    swipeDirection="down"
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      hideModalContentWhileAnimating={true}
      onModalHide={[Function]}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      pointerEvents="box-none"
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "left": 0,
          "margin": 37.5,
          "top": 0,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      swipeDirection="down"
      swipeThreshold={100}
    >
      <View />
    </View>
  </Modal>
</View>
`;
